contrib/packs/actions/pack_mgmt/get_pack_dependencies.py

```diff
@@ -0,0 +1,133 @@
+        """
+        :param packs_status: Name of the pack in Exchange or a git repo URL and download status.
+        :type: packs_status: ``dict``
+
+        :param nested: Nested level of dependencies to prevent infinite or really
+        long download loops.
+        :type nested: ``integer``
+        """
+        result = {}
+
+        if not packs_status or nested == 0:
+            return result
+
+        for pack, status in six.iteritems(packs_status):
+            if 'success' not in status.lower():
+                continue
+
+            dependency_packs = get_dependency_list(pack)
+            if not dependency_packs:
+                continue
+
+            for dependency_pack in dependency_packs:
+                pack_and_version = dependency_pack.split(PACK_VERSION_SEPARATOR)
+                name_or_url = pack_and_version[0]
+                pack_version = pack_and_version[1] if len(pack_and_version) > 1 else None
+
+                if name_or_url.startswith(LOCAL_FILE_PREFIX):
+                    self.get_local_pack_dependencies(name_or_url, pack_version, dependency_pack)
+                elif (len(name_or_url.split('/')) == 1 and STACKSTORM_EXCHANGE_PACK not in
+                      name_or_url) or STACKSTORM_EXCHANGE_PACK in name_or_url:
+                    self.get_exchange_pack_dependencies(name_or_url, pack_version, dependency_pack)
+                else:
+                    self.get_none_exchange_pack_dependencies(name_or_url, pack_version,
+                                                             dependency_pack)
+
+        result['dependency_list'] = self.dependency_list
+        result['conflict_list'] = self.conflict_list
+        result['nested'] = nested - 1
+
+        return result
+
+    # For StackStorm Exchange packs. E.g.
+    # email
+    # https://github.com/StackStorm-Exchange/stackstorm-email
+    # https://github.com/StackStorm-Exchange/stackstorm-email.git
+    def get_exchange_pack_dependencies(self, name_or_url, pack_version, dependency_pack):
+        if len(name_or_url.split('/')) == 1:
+            pack_name = name_or_url
+        else:
+            name_or_git = name_or_url.split(STACKSTORM_EXCHANGE_PACK)[-1]
+            pack_name = name_or_git if '.git' not in name_or_git else name_or_git.split('.')[0]
+
+        existing_pack_version = get_pack_version(pack_name)
+        self.check_conflict(dependency_pack, pack_version, existing_pack_version)
+
+    # For None StackStorm Exchange packs. E.g.
+    # https://github.com/EncoreTechnologies/stackstorm-freeipa.git
+    # https://github.com/EncoreTechnologies/stackstorm-freeipa
+    # https://github.com/EncoreTechnologies/pack.git
+    def get_none_exchange_pack_dependencies(self, name_or_url, pack_version, dependency_pack):
+        name_or_git = name_or_url.split('/')[-1]
+        name = name_or_git if '.git' not in name_or_git else name_or_git.split('.')[0]
+        pack_name = name.split('-')[-1] if "stackstorm-" in name else name
```

Comment:
In 3rd party Github URLs assumption that * in `stackstorm-*` would be always a pack name is bad and will eventually explode.

Reply:
Will rework on parse pack input and without make any assumptions.